controllers/datafoundation.go

```diff
@@ -287,7 +298,8 @@ func (r *dataFoundationReconciler) reconcileStorageCluster() error {
 			ds = desiredStorageDeviceSet
 		}
 
-		// Prevent downscaling by comparing count from secret and count from storage cluster
+		// Prevent downscaling by comparing OSD size and count from secret and from storage cluster
+		r.setDeviceSetOSDSize(ds, desiredOSDSizeInTiB, currOSDSizeInTiB)
```

Comment:
I don't see a point in having this separate from `setDevicesetCount`. Mainly as we need to protect against downscale as a result of these pair of parameters configured together. 

You might want to rename `setDeviceSetCount` to `setDeviceSetProperties` and have the entire logic in one place

Reply:
I was about to